### ISSUE TITLE
Disappearing messages disappear before I read them

### ISSUE THREAD

**Issue by Deadlocklucifer** [signalapp/Signal-iOS#1901]
Ios 10.2.2 on iPhone 7
Error occurs since call kit integration a month or so ago. 

When I leave the app while I'm a conversation screen message countdown automatically starts. 
Default expiration for me is 5 mins
If I go to the app more than 5 mins after the message was received it is no longer there. 

If however I leave the app when in the chat list view of all those I'm communicating with and a message comes in, the timer does not start. 

Also sometimes the notification for messages doesn't show up as a red bubble with number on app icon. But the phone will make a sound or vibrate.

**Comment by michaelkirk** [comment]
> When I leave the app while I'm a conversation screen message countdown automatically starts.
Default expiration for me is 5 mins
If I go to the app more than 5 mins after the message was received it is no longer there.
> 
> If however I leave the app when in the chat list view of all those I'm communicating with and a message comes in, the timer does not start.

It sounds like this is working as intended.

We don't start the countdown for disappearing messages until we're confident a message has been read, which doesn't happen until you tap into a thread.

If we started the disappearing messages timer before a user taps into the thread, I believe there would be many users who never actually saw the message, and were just scrolling through their thread list. The message would disappear before they ever actually saw it.

> Also sometimes the notification for messages doesn't show up as a red bubble with number on app icon. But the phone will make a sound or vibrate.

This sounds like a separate issue. Can you provide logs immediately after this happens? (Settings > Advanced > Submit debug logs).

**Comment by Deadlocklucifer** [comment]
Thanks Michael, I think you misunderstood or perhaps I did not explain properly, 

when I get messages they expire without me knowing they are there or without me even opening the app

it occurs when I leave the app in the conversation window and exit

i.e. the countdown starts prior to me even reading the message.

**Comment by michaelkirk** [comment]
> Thanks Michael, I think you misunderstood or perhaps I did not explain properly,

I guess I did misunderstand. Following the issue template would help with this. In particular being clear on what you *did* see vs what you *expected* to see.

So I think you're saying:

I have a conversation with Alice, with whom I have disappearing messages enabled (disappears after 5 mins).
Signal-iOS is open to my conversation with alice.
I leave the app (e.g. hit the home button)
Alice messages me "hello" while Signal is not active on my phone.
5 minutes later I open the app, and I see the conversation thread with Alice...

**What actually happened:**

But I never see that Alice messaged me "hello".

**What I expected to happen:**

The "Hello" message from Alice, with a just-started timer. 
5 minutes later, the message should be deleted.

Is that right?

**Comment by Deadlocklucifer** [comment]
Correct, far more eloquent than i.

**Comment by 8tq** [comment]
Hey everyone, thanks for the information and tracking this issue, as I find it an important factor impacting my usage of Signal. Has there been any progress or updates?

**Comment by michaelkirk** [comment]
@8tq - have you experienced something similar? Can you provide some details?

**Comment by ja3111** [comment]
This is a major problem for me right now. I use signal because of its claimed security and privacy functions. I use disappearing messages. Recently, I have set disappearing messages on shorter timeframes, say 5 seconds or 1 minute. I send a message to X, but the message has only one tick - which means that the person is not connected to the internet, have not received the message, and have not read the message until the following day. YET, the message disappears within 5 seconds or 1 minute of me sending it, NOT of them reading it. I tried this on both iPhone and Andriod, and experienced the same problem. Two different devices, two difference numbers, two different recipients. Is there a glitch in the app? Or should I be worried that my phone is hacked at the root? It can't be for both phones though, because the later is not even mine... PLEASE HELP!

**Comment by michaelkirk-signal** [comment]
@ja3111  - we start the timer for your phone when *you* see the message. We start the timer for the recipients copy when *they* see the message. It sounds like in your case, it's working as we intended.

**Comment by ja3111** [comment]
OK so just a few weeks back, the way it worked on my phone was that: when I sent a message, it would disappear within X seconds but only after the other person had seen it. It never used to disappear immediately from my phone even if the other person hadn't seen it yet.... So are you saying that now the message that I sent is deleted from my phone within X seconds of me sending it AND from the other person's phone within X seconds of them reading it?

**Comment by michaelkirk-signal** [comment]
We start the timer for your phone when you see the message. We start the timer for the recipients copy when they see the message.

This is the way disappearing messages have *always* worked on Signal, for iOS, Android, and Desktop.

**Comment by Jsnkenk** [comment]
Seems like this issue is not fixed i am facing the same issue for 5min auto disappear in android 11 on pixel 4

**Comment by eteldig** [comment]
Just downloaded this app and set up disappearing messages. I’m still trying to understand how they work. But I’m fairly certain based on a few different time settings I’m seeing how it works?

I’ve had an hour timer and a day so far set. 
I noticed regardless of that, I think it goes based on EACH person’s device and when THEY open it too. I don’t think it just disappears entirely for both, even if unread by the recipient? 

Example, I received one message the previous day from someone, in the morning (1 day timer set for the convo). They sent a message at 5am yesterday. I read/opened this at 10am yesterday, and I responded. After 24 hours of MYSELF opening it, all messages disappeared today at 10am on my phone. I responded yesterday and it was delivered. But I can’t see now if they have read it. But I’m assuming it’s still on their device until THEY open the app and tap into the thread? So now it’s gone from my own device but likely not there’s? And I’m pretty sure this is how this should work?

**Comment by ralphstone1** [comment]
IDK

**Comment by atljoseph** [comment]
"When enabled, new messages sent and received in this chat will disappear after they have been seen"

That is the message the app displays at the top of the Disappearing Messages configuration screen. It implies that messages disappear after both parties see it (plus a certain a amount of time after that point). 

This is not what happens, however. What happens is as described in previous comments in this thread. 

It is a shame that the app's verbiage doesn't help the user understand how it works. 

It is another shame that the person reporting the issue has a GREAT point with GREAT feedback, and the person triagibg the issue had ZERO tact in dealing with the issue, and extracted NONE of the great feedback yielded. 

@Deadlocklucifer, I'm with ya. Why doesnt the app say one thing in this config screen and act differently in reality??? They could treat the issue with some more reverence.

**Comment by atljoseph** [comment]
If you make the user less confused, the user will be less confused. Voila.

**Comment by atljoseph** [comment]
A conversation exists in 2 users' devices. Unless explicitly mentioning that the messages disappear from each device at an independent time, you are telling the user implicitly that it disappears from all devices simultaneously. Think about it.
